tests/linearizability/watch.go

```diff
@@ -47,53 +47,90 @@ func collectClusterWatchEvents(ctx context.Context, t *testing.T, lg *zap.Logger
 		go func(i int, c *clientv3.Client) {
 			defer wg.Done()
 			defer c.Close()
-			events := collectMemberWatchEvents(ctx, lg, c)
+			responses := watchMember(ctx, lg, c)
 			mux.Lock()
-			memberEvents[i] = events
+			memberResponses[i] = responses
 			mux.Unlock()
 		}(i, c)
 	}
 	wg.Wait()
-	return memberEvents
+	return memberResponses
 }
-
-func collectMemberWatchEvents(ctx context.Context, lg *zap.Logger, c *clientv3.Client) []watchEvent {
-	events := []watchEvent{}
-	var lastRevision int64 = 1
+func watchMember(ctx context.Context, lg *zap.Logger, c *clientv3.Client) (resps []watchResponse) {
+	var lastRevision int64 = 0
 	for {
 		select {
 		case <-ctx.Done():
-			return events
+			return resps
 		default:
 		}
-		for resp := range c.Watch(ctx, "", clientv3.WithPrefix(), clientv3.WithRev(lastRevision)) {
+		for resp := range c.Watch(ctx, "", clientv3.WithPrefix(), clientv3.WithRev(lastRevision+1)) {
+			resps = append(resps, watchResponse{resp, time.Now()})
 			lastRevision = resp.Header.Revision
-			time := time.Now()
-			for _, event := range resp.Events {
-				var op model.OperationType
-				switch event.Type {
-				case mvccpb.PUT:
-					op = model.Put
-				case mvccpb.DELETE:
-					op = model.Delete
-				}
-				events = append(events, watchEvent{
-					Time:     time,
-					Revision: event.Kv.ModRevision,
-					Op: model.EtcdOperation{
-						Type:  op,
-						Key:   string(event.Kv.Key),
-						Value: model.ToValueOrHash(string(event.Kv.Value)),
-					},
-				})
-			}
 			if resp.Err() != nil {
 				lg.Info("Watch error", zap.Error(resp.Err()))
 			}
 		}
 	}
 }
 
+func validateWatchResponses(t *testing.T, responses [][]watchResponse) {
+	for _, memberResponses := range responses {
+		validateMemberWatchResponses(t, memberResponses)
+	}
+}
+
+func validateMemberWatchResponses(t *testing.T, responses []watchResponse) {
+	var lastRevision int64 = 1
+	for _, resp := range responses {
+		if resp.Header.Revision < lastRevision {
+			t.Errorf("Revision should never decrease")
```

Comment:
This minor comment applies to the following couple of places (`t.Errorf`)
```suggestion
			t.Error("Revision should never decrease")
```